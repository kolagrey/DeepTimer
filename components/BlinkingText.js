import React, { Component } from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';

class BlinkingText extends Component {
  constructor(props) {
    super(props);
    this.state = { show: true };

    this.chronos = setInterval(() => {
        this.setState(previousState => {
          return { show: !previousState.show };
        });
      },500);

  }

  componentWillUnmount() {
    clearInterval(this.chronos);
  }

  render() {
    const { style, text } = this.props;
    const { show } = this.state;
    const displayText = show ? text : '';
    return (
      <Text style={style}>{displayText}</Text>
    );
  }
}

BlinkingText.propTypes = {
  style: PropTypes.object,
  text: PropTypes.string
}

export default BlinkingText;