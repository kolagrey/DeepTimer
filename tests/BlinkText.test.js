import React from 'react';
import renderer from 'react-test-renderer';

import BlinkingText from '../components/BlinkingText';


describe('<BlinkingText/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<BlinkingText />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'Text'", () => {
        const tree = renderer
            .create(
                <BlinkingText />,
            ).toJSON();
        expect(tree.type).toBe('Text');
    });

    it('displays the text property passed', () => {
        const tree = renderer.create(<BlinkingText text="Blink" />).toJSON();
        expect(tree.children[0]).toBe('Blink');
    });

    it('has a default state show = true', () => {
        const tree = renderer.create(<BlinkingText text="Blink" />);
        expect(tree.root.instance.state.show).toBe(true);
    });

});