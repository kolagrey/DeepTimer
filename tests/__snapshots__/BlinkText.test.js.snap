// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlinkingText/> renders correctly 1`] = `<Text />`;
